crates/iota-framework/packages/iota-system/tests/rewards_distribution_tests.move

```diff
@@ -972,11 +974,137 @@ module iota_system::rewards_distribution_tests {
+        advance_epoch_with_reward_amounts(0, 3, scenario);
+        advance_epoch_with_reward_amounts(0, 1, scenario);
+       
+       // Epoch with addition of stake and no rewards
+        stake_with(STAKER_ADDR_3, VALIDATOR_ADDR_1, 3, scenario);
+        advance_epoch(scenario);
+
+       // Epoch with removal of stake and no rewards
+        unstake(STAKER_ADDR_2, 0, scenario);
+        advance_epoch(scenario);
+
+        // Epoch with rewards
+        advance_epoch_with_reward_amounts(0, 100, scenario);
+
+        // Get exchange rates for the pool to check its pool token supply during those epoch changes
+        scenario.next_tx(VALIDATOR_ADDR_1);
+        let mut system_state = scenario.take_shared<IotaSystemState>();
+
+        let staked_iota = scenario.take_from_address<StakedIota>(STAKER_ADDR_1);
+        let pool_id = staked_iota.pool_id();
+        let rates = system_state.pool_exchange_rates(&pool_id);
+        let pool_token_amount_epoch_3  = rates[3].pool_token_amount() as u128;
+        let pool_token_amount_epoch_4  = rates[4].pool_token_amount() as u128;
+        let iota_amount_epoch_4  = rates[4].iota_amount() as u128;
+        let pool_token_amount_epoch_5  = rates[5].pool_token_amount() as u128;
+        let pool_token_amount_epoch_6  = rates[6].pool_token_amount() as u128;
+        test_scenario::return_to_address(STAKER_ADDR_1, staked_iota);
+
+        // Test 1: from epoch 3 to 4, 3_000_000_000 NANOs were added to the pool. The number of pool tokens
+        // minted should be equal to 3_000_000_000 * pool_token_amount_epoch_4 / iota_amount_epoch_4, in theory.
+        // Because of the fixed point arithmetic, the result is not exact, so we accept a small error.
+        assert!(iota_amount_epoch_4 * (pool_token_amount_epoch_4 - pool_token_amount_epoch_3) > (3_000_000_000 - 1_000) * pool_token_amount_epoch_4, 0);
+        assert!(iota_amount_epoch_4 * (pool_token_amount_epoch_4 - pool_token_amount_epoch_3) < (3_000_000_000 + 1_000) * pool_token_amount_epoch_4, 0);
+
+        // Test 2: from epoch 4 to 5, 50_000_000_000 NANOs plus its rewards were removed from the pool. 
+        // The number of burned pool tokens should be equal to 50_000_000_000, since this stake was added 
+        // in the first epoch, when we had 1 pool token per NANO.
+        assert!(pool_token_amount_epoch_4  - pool_token_amount_epoch_5 == 50_000_000_000, 0);
+
+        // Test 3: from epoch 5 to 6, no IOTAs were added or removed from the pool. 
+        // The number of pool tokens should be constant even with the distribution of rewards.
+        assert!(pool_token_amount_epoch_5 == pool_token_amount_epoch_6, 0);
+
+        test_scenario::return_shared(system_state);
+        scenario_val.end();
+    }
+
     // This will set up the IOTA system state with the following validator stakes:
-    // Valdiator 1 => 100
-    // Valdiator 2 => 200
-    // Valdiator 3 => 300
-    // Valdiator 4 => 400
+    // Validator 1 => 100
```

Comment:
I would suggest another alternative spelling, "Vladiator" which kind of sounds cooler like some sort of Slavic gladiator. Joke obviously...

Reply:
Yes, you corrected the spelling, I was just making a bad joke...